compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt

```diff
@@ -117,8 +118,32 @@ val FirClassSymbol<*>.superConeTypes
 
 val FirClass<*>.superConeTypes get() = superTypeRefs.mapNotNull { it.coneTypeSafe<ConeClassLikeType>() }
 
-fun FirClass<*>.getPrimaryConstructorIfAny(): FirConstructor? =
-    declarations.filterIsInstance<FirConstructor>().firstOrNull()?.takeIf { it.isPrimary }
+val FirClass<*>.anonymousInitializers: List<FirAnonymousInitializer>
+    get() = declarations.filterIsInstance<FirAnonymousInitializer>()
+
+val FirClass<*>.constructors: List<FirConstructor>
+    get() = declarations.filterIsInstance<FirConstructor>()
+
+val FirConstructor.delegatedThisConstructor: FirConstructor?
+    get() = delegatedConstructor?.takeIf { it.isThis }
+        ?.let { (it.calleeReference as? FirResolvedNamedReference)?.resolvedSymbol?.fir as? FirConstructor }
+
+private object ConstructorDelegationComparator : Comparator<FirConstructor> {
+    override fun compare(p0: FirConstructor?, p1: FirConstructor?): Int {
+        if (p0 == null && p1 == null) return 0
+        if (p0 == null) return -1
+        if (p1 == null) return 1
+        if (p0.delegatedThisConstructor == p1) return 1
+        if (p1.delegatedThisConstructor == p0) return -1
+        return p0.hashCode() - p1.hashCode()
```

Comment:
IIUC then the order shouldn't matter here when they're not delegating to each other. In that case maybe you can just return 1 to preserve the original order (hashCode will result in non-deterministic order).

Reply:
To preserve the original order, shouldn't 0 be returned?

Reply:
Oops yes I stand corrected

Reply:
Right, the order doesn't matter for the unrelated constructors. I didn't like `hashCode` calls anyway, so sounds good, thanks!